Userland/Libraries/LibAudio/Metadata.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/HashMap.h>
+#include <AK/LexicalPath.h>
+#include <AK/Optional.h>
+#include <AK/String.h>
+#include <AK/Time.h>
+#include <AK/Variant.h>
+
+namespace Audio {
+
+struct Person {
+    enum class Role {
+        Artist,
+        Performer,
+        Lyricist,
+        Conductor,
+        Publisher,
+        Engineer,
+        Composer,
+    };
+    Role role;
+    String name;
+
+    // Whether this person has creative involvement with the song (so not only Role::Artist!).
+    // This list is subjective and is intended to keep the artist display text in applications relevant.
+    // It is used for first_artist and all_artists in Metadata.
+    bool is_artist() const;
+
+    Optional<StringView> name_for_role() const;
+};
+
+// Audio metadata of the original format must be equivalently reconstructible from this struct.
+// That means, (if the format allows it) fields can appear in a different order, but all fields must be present with the original values,
+// including duplicate fields where allowed by the format.
+struct Metadata {
+    using Year = unsigned;
+
+    Metadata() = default;
+    Metadata(Metadata const&) = default;
+    Metadata(Metadata&&) = default;
+    ~Metadata() = default;
+    Metadata& operator=(Metadata const&) = default;
+    Metadata& operator=(Metadata&&) = default;
```

Comment:
Since all of these are =default, you could just not declare them at all

Reply:
Yeah, I wanted to make sure that they exist, as the rules for when they are deleted are not super obvious. But here it should be fine.